src/modules/form/components/DynamicFormField.tsx

```diff
@@ -0,0 +1,276 @@
+import { formatCurrency } from '@/modules/hmis/hmisUtil';
+import {
+  Component,
+  FormItem,
+  ItemType,
+  ServiceDetailType,
+} from '@/types/gqlTypes';
+
+export const renderItemWithWrappers = (
+  renderChild: (disabled?: boolean) => ReactNode,
+  item: FormItem,
+  handlers: FormDefinitionHandlers
+) => {
+  const { disabledDependencyMap, autofillInvertedDependencyMap } = handlers;
+
+  if (item.hidden) return null;
+
+  const hasDependencies =
+    disabledDependencyMap[item.linkId] ||
+    !isEmpty(item.enableWhen) ||
+    item.item?.every((item) => item.enableWhen);
+  const hasAutofill =
+    autofillInvertedDependencyMap[item.linkId] || !isEmpty(item.autofillValues);
+
+  if (hasDependencies && hasAutofill) {
+    return (
+      <DependentFormItemWrapper handlers={handlers} item={item}>
+        {(disabled) => (
+          <AutofillFormItemWrapper handlers={handlers} item={item}>
+            {() => renderChild(disabled)}
+          </AutofillFormItemWrapper>
+        )}
+      </DependentFormItemWrapper>
+    );
+  }
+
+  if (hasDependencies) {
+    return (
+      <DependentFormItemWrapper handlers={handlers} item={item}>
+        {renderChild}
+      </DependentFormItemWrapper>
+    );
+  }
+
+  if (hasAutofill) {
+    return (
+      <AutofillFormItemWrapper handlers={handlers} item={item}>
+        {() => renderChild()}
+      </AutofillFormItemWrapper>
+    );
+  }
+
+  return renderChild();
+};
+
+export interface Props {
+  handlers: FormDefinitionHandlers;
+  clientId?: string;
+  horizontal?: boolean;
+  warnIfEmpty?: boolean;
+  locked?: boolean;
+  bulk?: boolean;
+  visible?: boolean;
+  pickListArgs?: PickListArgs;
+  nestingLevel: number;
+  item: FormItem;
+  props?: OverrideableDynamicFieldProps;
+  renderFn?: (children: ReactNode) => ReactNode;
+}
+
+const DynamicFormField: React.FC<Props> = ({
+  handlers,
+  clientId,
+  bulk,
+  horizontal = false,
+  warnIfEmpty = false,
+  locked = false,
+  visible = true,
+  pickListArgs,
+  item,
+  nestingLevel,
+  props: fieldProps,
+  renderFn,
+}) => {
+  const values = handlers.getCleanedValues();
+  const { definition, localConstants, getFieldErrors } = handlers;
+
+  const itemChanged: ItemChangedFn = useCallback(
+    ({ linkId, value: baseValue, type }) => {
+      const item = handlers.itemMap[linkId];
+      let value = baseValue;
+      if (item) {
+        if (item.type === ItemType.Integer) value = parseInt(value) || 0;
+        if (item.type === ItemType.Currency) value = parseFloat(value) || 0;
+      }
+      handlers.methods.setValue(getSafeLinkId(linkId), value, {
+        shouldDirty: type === ChangeType.User,
+      });
+    },
+    [handlers]
+  );
+  const severalItemsChanged: SeveralItemsChangedFn = useCallback(
+    ({ values, type }) => {
+      Object.entries(values).forEach(([linkId, value]) =>
+        handlers.methods.setValue(getSafeLinkId(linkId), value, {
+          shouldDirty: type === ChangeType.User,
+        })
```

Comment:
Maybe dry this up and use itemChanged so we get parseInt/Float?
```suggestion
        itemChanged({linkId, value, type})
```